fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java

```diff
@@ -623,6 +623,41 @@ private AppUser getAppUserIfPresent() {
         return user;
     }
 
+    @Override
+    public CommandProcessingResultBuilder creditBalanceRefund(Long loanId, LocalDate transactionDate, BigDecimal transactionAmount,
```

Comment:
Same here just like with the Loan class. This is also getting too big.

I could imagine this LoanAccountDomainServiceJpa class to be kind of a facade and have a specific CreditBalanceRefundService class which has this implementation only and the LoanAccountDomainServiceJpa class just delegates the call.
Thoughts?